yt_dlp/extractor/bbc.py

```diff
@@ -1191,6 +1279,28 @@ def _real_extract(self, url):
                 return self.playlist_result(
                     entries, playlist_id, playlist_title, playlist_description)
 
+        k_int_or_none = functools.partial(int_or_none, scale=1000)
+
+        def parse_model(model):
+            '''Extract single video from model structure'''
+            item_id = traverse_obj(model, ('versions', 0, 'versionId', {str}))
+            if not item_id:
+                return
+            formats, subtitles = self._download_media_selector(item_id)
+            return {
+                'id': item_id,
+                'formats': formats,
+                'subtitles': subtitles,
+                **traverse_obj(model, {
+                    'title': ('title', {str}),
+                    'thumbnail': ('imageUrl', {lambda u: urljoin(url, u.replace('$recipe', 'raw'))}),
+                    'description': (
+                        'synopses', ('long', 'medium', 'short'), {str}, any),
+                    'duration': ('versions', 0, 'duration', {int}),
+                    'timestamp': ('versions', 0, 'availableFrom', {k_int_or_none}),
```

Comment:
this was made as a function because is conversion is used twice. I have tried both, and this lambda and the functools.partial are equivalent in functionality.

Reply:
Side note, `partial` should always be preferred imo. It will give a more clear indication of which function was called in tracebacks

Reply:
```suggestion
                    'timestamp': ('versions', 0, 'availableFrom', {functools.partial(int_or_none, scale=1000)}),
```